terraform/modules/asg/main.tf

```diff
@@ -0,0 +1,156 @@
+##
+# Data providers
+##
+data "aws_vpc" "selected" {
+  id = "${var.vpc_id}"
+}
+
+data "aws_subnet_ids" "app" {
+  vpc_id = "${var.vpc_id}"
+
+  tags {
+    Layer       = "app"
+    stack       = "${var.stack}"
+    application = "${var.app}"
+  }
+}
+
+data "aws_subnet" "app" {
+  count = "${length(data.aws_subnet_ids.app.ids)}"
+  id    = "${data.aws_subnet_ids.app.ids[count.index]}"
+}
+
+data "aws_elb" "elb" {
+  name = "${var.elb_name}"
+}
+
+##
+# Security groups
+##
+resource "aws_security_group" "ci" {
+  name        = "ci-to-app-servers"
+  description = "Allow CI access to app servers"
+  vpc_id      = "${data.aws_vpc.selected.id}"
+
+  ingress {
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = ["${var.ci_cidrs}"]
+  }
+}
+
+##
+# Launch configuration
+##
+resource "aws_launch_configuration" "app" {
+  security_groups = [
+    "${var.app_sg_id}",
+    "${var.vpn_sg_id}",
+    "${aws_security_group.ci.id}",
+  ]
+
+  key_name                    = "${var.key_name}"
```

Comment:
I figure this is worth asking, though not a major concern given the time constraints.  Where is this key stored?  What level of access does it give to the machines (I know if nothing else is done to the account it's root, but wondering if that is true even after initialization).